homeassistant/components/sensor/dte_energy_bridge.py

```diff
@@ -91,4 +91,9 @@ def update(self):
                 response.text, self._name)
             return
 
-        self._state = float(response_split[0])
+        val = float(response_split[0])
+
+        # A workaround for a bug in the DTE energy bridge.
+        # The returned value can randomly be in W or kW.  Checking for a
+        # a decimal seems to be a reliable way to determine the units.
+        self._state = val if '.' in response_split[0] else val / 1000
```

Comment:
Can we fix that by provider of that data?

Reply:
For two reasons, I'm guessing not.

1.  The API is undocumented.
2.  A company (Powerley) was contracted to build this device by DTE Energy.  I'm guessing it would probably be a bureaucratic mess to request changes.

Also, this is their V1 of the device.  Hoping at this point that their V2 device won't have this problem.  (PR coming soon to support V2)